internal/store/pod.go

```diff
@@ -1071,7 +1072,7 @@ func waitingReason(cs v1.ContainerStatus, reason string) bool {
 	if cs.State.Waiting == nil {
 		return false
 	}
-	return cs.State.Waiting.Reason == reason
+	return cs.State.Waiting.Reason == reason || strings.Contains(cs.State.Waiting.Reason, "PostStartHookError")
```

Comment:
@pigletfly I am not very comfortable with this logic since we're hardcoding `PostStartHookError` here. Thoughts @brancz and @LiliC ?

Reply:
Yeah I'm not a fan of this either, it leaks into some weird hybrid of something like log analysis I feel. I don't think this belongs in kube state metrics unless it becomes a standardized API.